### PR TITLE
Add igwn-lldd-common and igwn-lldd-fdio

### PR DESCRIPTION
**Pull request by rhyspoulton**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rhyspoulton** [comment]
@duncanmmacleod two questions:

1.  Are you happy to be added as a maintainer of this recipe?
2. Do you know why the `win_64` check failed? : https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=577555&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf

**Comment by duncanmmacleod** [comment]
> @duncanmmacleod two questions:
> 
> 1. Are you happy to be added as a maintainer of this recipe?

Yes, I am happy to be named.

> 2. Do you know why the `win_64` check failed? : https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=577555&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf

I _think_ it's because you haven't declared the entry points in the conda recipe.

File: recipes/igwn-lldd-common/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{% set name = "igwn-lldd-common" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://software.igwn.org/lscsoft/source/igwn-lldd-common-0.1.0.tar.gz
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-common/meta.yaml:9]: It's probably best to use the jinja2 variables here:

```suggestion
  url: https://software.igwn.org/lscsoft/source/{{ name }}-{{ version }}.tar.gz
```

File: recipes/igwn-lldd-common/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  sha256: fa70188e6185e807c92fd10f2b82bcd6046435148aebda18fb3a0f8b419a6ef9
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vvv
+  run_exports:
+    - {{ pin_subpackage("igwn-lldd-common", max_pin="x") }}
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-common/meta.yaml:17]: I've never seen a pure python package provide `run_exports`, I'm not sure this is needed here.

File: recipes/igwn-lldd-common/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - configargparse
+    - ujson
+    - gpstime
+    - watchdog
+
+test:
+  requires:
+    - pip
+    - pytest
+    - pytest-cov
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-common/meta.yaml:37]: Measuring coverage in the conda tests is overkill, probably remove.

```suggestion
```

File: recipes/igwn-lldd-common/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    - python -m pip check
+    # run test suite
+    - python -m pytest --cov=igwn_lldd_common -ra -v --pyargs igwn_lldd_common.tests
+    # check entry points
+    - python -m coverage run --append --source=igwn_lldd_common -m igwn_lldd_common.frame2kafka --help
+    - python -m coverage run --append --source=igwn_lldd_common -m igwn_lldd_common.kafka2frame --help
+    # print out the coverage
+    - python -m coverage report
+    # put coverage in a xml file
+    - python -m coverage xml -o coverage-conda.xml
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-common/meta.yaml:50]: As above, measuring coverage is unnecessary (also use the entry points)

```suggestion
    - python -m pytest -ra -v --pyargs igwn_lldd_common.tests
    # check entry points
    - frame2kafka --help
    - kafka2frame --help
```

File: recipes/igwn-lldd-common/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    - python >=3.6
+    - setuptools
+  run:
+    - python >=3.6
+    - python-confluent-kafka
+    - kafka-python
+    - configargparse
+    - ujson
+    - gpstime
+    - watchdog
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-common/meta.yaml:31]: Can you alphabetise these to make maintenance easier?

File: recipes/igwn-lldd-common/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - setuptools
+  run:
+    - python >=3.6
+    - python-confluent-kafka
+    - kafka-python
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-common/meta.yaml:27]: As we discussed privately, I think you only need `python-confluent-kafka` here (to match the Python package), but support for the other kafka implementation should be documented (not here).
```suggestion
    - python-confluent-kafka
```

Comment (rhyspoulton) [reply recipes/igwn-lldd-common/meta.yaml]: I have discussed this with Patrick and he said that `python-confluent-kafka` is already the case for `RPMs`

File: recipes/igwn-lldd-common/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    - pytest
+  commands:
+    # check metadata
+    - python -m pip check igwn-lldd-common
+    - python -m pip show igwn-lldd-common
+    # run test suite
+    - python -m pytest -ra -v --pyargs igwn_lldd_common.tests
+    # check entry points
+    - python -m igwn_lldd_common.frame2kafka --help
+    - python -m igwn_lldd_common.kafka2frame --help
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-common/meta.yaml:47]: I think here you want to actually check the entry points as declared in `build`:

```suggestion
    - frame2kafka --help
    - kafka2frame --help
```

File: recipes/igwn-lldd-fdio/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    - flaky
+  commands:
+    # check metadata
+    - python -m pip check igwn-lldd-fdio
+    - python -m pip show igwn-lldd-fdio
+    # run test suite
+    - python -m pytest -ra -v --pyargs igwn_lldd_fdio.tests
+    # check entry points
+    - python -m igwn_lldd_common.fdio2kafka --help
+    - python -m igwn_lldd_common.kafka2fdio --help
```
Comment (duncanmmacleod) [inline recipes/igwn-lldd-fdio/meta.yaml:45]: As above (I think the library name is wrong anyway):

```suggestion
    - fdio2kafka --help
    - kafka2fdio --help
```

**Comment by rhyspoulton** [comment]
@conda-forge/help-python , this recipe is ready for review. The OSX and WIN build fail for `igwn-lldd-fdio` due to `pyfd` and `pythonvirgotools` not being available in these distributions.

**Comment by duncanmmacleod** [comment]
@conda-forge/staged-recipes, is someone able to review this recipe?

As @rhyspoulton notes the non-linux builds fail because of dependency issues, but the packages are all `noarch:python`.

**Comment by duncanmmacleod** [comment]
Ping @conda-forge/staged-recipes, please review, thanks.

File: recipes/igwn-lldd-fdio/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    - python -m pip check igwn-lldd-fdio
+    - python -m pip show igwn-lldd-fdio
+    # run test suite
+    - python -m pytest -ra -v --pyargs igwn_lldd_fdio.tests
+    # check entry points
+    - fdio2kafka --help
+    - kafka2fdio --help
+
+about:
+  home: "https://git.ligo.org/computing/lowlatency/igwn-lldd-fdio"
```
Comment (BastianZim) [inline recipes/igwn-lldd-fdio/meta.yaml:48]: This link is private. Is there one where I can see the project?

Comment (duncanmmacleod) [reply recipes/igwn-lldd-fdio/meta.yaml:48]: @bastianzim, this link (the relevant gitlab project) has been made publicly viewable.

**Pull request merged**